discovery-provider/src/premium_content/premium_content_access_checker.py

```diff
@@ -37,12 +40,28 @@ class PremiumContentAccessBatchResponse(TypedDict):
     track: PremiumTrackAccessResult
 
 
+class PremiumContentAccessHandler(Protocol):
+    def __call__(
+        self,
+        session: Session,
+        user_id: int,
+        content_id: int,
+        content_type: PremiumContentType,
+        condition_options: dict,
```

Comment:
confused as to this being a dict here, but in the helper functions above it's an int?

Reply:
goooood catch